### ISSUE TITLE
Add `ignores_literals` option to `trailing_whitespace` rule

### ISSUE THREAD

**Issue by iliaskarim** [realm/SwiftLint#6194]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Feature or Enhancement Proposal

Add a configuration option to the `trailing_whitespace` rule to **ignore trailing whitespace inside string literals**.

Example configuration:

```yaml
trailing_whitespace:
  ignores_literals: true
```

#### Rationale

- Trailing whitespace inside string literals can be **intentional** (for example, when storing patch strings, diffs, or preformatted text).  
- The current `trailing_whitespace` rule raises violations for these cases, requiring developers to disable the rule around literals or add per-line suppressions.  
- An `ignores_literals` option would allow teams to enforce trailing-whitespace rules for code while preserving meaningful whitespace inside literals.

#### Notes

- Default behavior should remain unchanged (i.e., `ignores_literals: false`) for backward compatibility.  
- Optionally, a future enhancement could add a finer-grained flag (e.g., only ignore in multiline literals), but this initial proposal keeps the API minimal and practical.
